src/components/pages/reusable-ui/Input.js

```diff
@@ -1,24 +1,81 @@
-import styled from 'styled-components'
-import { theme } from '../../../theme'
+import { useRef } from "react";
+import styled, { css } from "styled-components";
+import { theme } from "../../../theme";
 
-export default function Input({value,onChange,Icon, ...extraProps}) {
-
-    return (
-        <InputStyled>
-            <span className="icon">{Icon && Icon}</span>
-            <input
-                value={value}
-                onChange={onChange}
-                {...extraProps}
-            />
-        </InputStyled>
-    )
+export default function Input({
+  value,
+  onChange,
+  Icon,
+  placeholder,
+  variant = "normal",
+  ...extraProps
+}) {
+  const inputReference = useRef();
+  return (
+    <InputStyled variant={variant}>
+      <span className="icon">{Icon && Icon}</span>
```

Comment:
```suggestion
      {Icon && <span className="icon">{Icon}</span>}
```

Faut qu'on voit ça ensemble en vocal juste après, je te ré-expliquerai ça proprement 😉